verification/src/download_cache.rs

```diff
@@ -0,0 +1,189 @@
+use async_trait::async_trait;
+use semver::Version;
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+
+#[async_trait]
+pub trait Fetcher {
+    type Error: std::error::Error;
```

Comment:
```suggestion
    type Error;
```
Why should it implement `std::error::Error` and cannot be more general?